New grid layout

wielinde opened the pull request:
**Done:**
- [x] Gray background
- [x] Remove widget icons
- [x] Replace widget border by soft shadow
- [x] Reposition widget title and give it some letter spacing
- [x] Apply styles to homescreen, removing icons there, too
- [x] Remove grid configuration options on mobile

**TODO:**
- [ ] Harmonize page header and toolbar margins 
- [ ] Adjust distance between widget title and content per widget type.

**Questions:**
- [ ] What happens with the gray background when people created a "all white" theme  

https://community.openproject.com/projects/openproject/work_packages/31062

wielinde committed:
Set grid background color

wielinde committed:
Decrease and widen widget title font

wielinde committed:
Symetric top left margins for widget titles

wielinde committed:
Widget box shadow instead of border

wielinde committed:
Remove icons from widget header

wielinde committed:
Remove white background from mobile toolbar

Disclaimer: I don't understand why it had been there in the first place

wielinde committed:
Change widget titles to uppercase

wielinde committed:
Remove grid configuration on mobile

wielinde committed:
Reduce padding for widget cards

wielinde committed:
Use light gray grid background for homescreen

wielinde committed:
Remove icons from homescreen widget cards

wielinde committed:
Reapply widget box header styles to homescreen by generalizing them

Before they were trapped in a component's style

HDinger commented on app/assets/stylesheets/content/_widget_box.sass:
```diff
@@ -112,25 +112,23 @@ $widget-box--enumeration-width: 20px
-.widget-box--header-icon
-  align-self: center
-  line-height: 28px
+    font-size: 16px
```
As this are the same styles you apply for the input field [here](https://github.com/opf/openproject/blob/e5afd2f1cc3d9d74ccfaecd651d77ddfab62e5dc/app/assets/stylesheets/content/_grid.sass#L121) it makes sense to create a mixing for this. Thus we can be sure that both, editable and non-editable headers, will always look and feel the same.

wielinde replied:
I united them.

HDinger commented on app/assets/stylesheets/openproject/_homescreen.sass:
```diff
@@ -28,9 +28,10 @@
+.controller-homescreen
+  #content-wrapper,
+  #content
+    background: $grid-background-color
```
It is new for me that the changed background should also be applied to the start page. It probably makes sense, because otherwise you won't see the widgets well. However, there are a few other things to consider. For example, the footer doesn't stand out anymore. And some of the changes to the distances don't apply to the home page, which makes them look different again. At least the footer should be better visible again.

![Bildschirmfoto 2019-10-14 um 09 34 14](https://user-images.githubusercontent.com/7457313/66737105-5cf54500-ee6b-11e9-9c47-cc166f42d803.png)

ulferts replied:
The my page should also receive the gray background.

![image](https://user-images.githubusercontent.com/617519/66746647-b026c200-ee82-11e9-814c-d10a56a7281a.png)

HDinger replied:
👍

HDinger commented on frontend/src/app/modules/grids/widgets/header/header.component.html:
```diff
@@ -1,8 +1,5 @@
     [ngClass]="{ '-editable': isRenameable }">
-  <i class="widget-box--header-icon icon-context"
-     aria-hidden="true"
-     [ngClass]="iconClass"></i>
```
Please remove this method, as well as the input `icon` in the component. Further the `widgets/wp-table.component` still uses the `icon` - input, so please remove this one, too.

wielinde replied:
Thanks, solved.

HDinger commented on app/assets/stylesheets/content/_grid.sass:
```diff
@@ -1,7 +1,7 @@
 $grid--header-width: 20px
 
-$grid--widget-padding: 20px
+$grid--widget-padding: 11px 20px 20px 20px
```
I'm not so happy with all those magic numbers. I know that it is often a necessary evil to align things correctly. But it only gets worse when you add the spacing correction. Maybe you can look again if you can draw some numbers straight and the distance can't be 10px here for example. However this is has a rather low priority.

wielinde replied:
I just changed that to 20px on all sides. But that is actuallay a coincidence. After talking to Christoph we decided to give the title more space on top than on the other sides.

HDinger reviewed (changes requested):
I think there are still some tasks open regarding the contrast with the new background. I'll give you a few examples and put the screenshots below:

* The button on the overview page has a very bad contrast of only 1.02(!). This is not only for visually limited users much too little. I have to admit that I don't really know which color is the right one now. You'll probably have to talk to Christoph again.
![Bildschirmfoto 2019-10-14 um 09 40 05](https://user-images.githubusercontent.com/7457313/66738153-1ead5500-ee6e-11e9-88a4-58dea23e2b30.png)
![Bildschirmfoto 2019-10-14 um 09 39 54](https://user-images.githubusercontent.com/7457313/66738154-1ead5500-ee6e-11e9-891b-dcd8d87f9785.png)

* When resizing and dragging, the other widgets are also very bad to recognize with a contrast of 1.06. 
![Bildschirmfoto 2019-10-14 um 09 38 14](https://user-images.githubusercontent.com/7457313/66738165-253bcc80-ee6e-11e9-8cee-e2768d061fa6.png)
![Bildschirmfoto 2019-10-14 um 09 37 23](https://user-images.githubusercontent.com/7457313/66738166-253bcc80-ee6e-11e9-8f48-fe2800324717.png)

* The same applies to the plus areas. The areas between the widgets still have a strange white border, which doesn't help it either.
![Bildschirmfoto 2019-10-14 um 09 39 44](https://user-images.githubusercontent.com/7457313/66738226-400e4100-ee6e-11e9-8b13-a0789b2a9048.png)
![Bildschirmfoto 2019-10-14 um 09 39 16](https://user-images.githubusercontent.com/7457313/66738280-574d2e80-ee6e-11e9-933a-0bf54f33b01b.png)

* Even the scrollbar is affected. Even if it still has the "best" contrast with 1.25, and is certainly the least important of the examples, I noticed it.
![Bildschirmfoto 2019-10-14 um 09 39 44 Kopie](https://user-images.githubusercontent.com/7457313/66738316-764bc080-ee6e-11e9-852f-caede0768824.png)

HDinger commented:
@wielinde I just had a look at our contrasts with the white background, and there are equally bad 😅 However I suggest, you have a look at the home screen footer, as well as the dragging and resizing styles. As far as I know, the grey background is not desired anyway for the widgets. Maybe you can remove the white border from the plus-areas, too. Leave the rest (scrollbar and the button) as they are.

wielinde commented:
Thank you @HDinger!

wielinde committed:
Use gray grid-background for my-page, too.

wielinde committed:
Fix: Remove widget icon input left overs

wielinde committed:
Unify duplicated styles

wielinde committed:
Make homescreen desing distinct from dashboard

also improve footer slightly slightly by removing box and applying the hover color to the icon, too.

wielinde committed:
Remove white borders from grid panels for adding widgets

wielinde committed:
Increase padding top of widget boxes to achieve asymmetric title margins

wielinde commented:
I solved the issue with white borders of the "add widget panels/tabs" by removing the border alltogether. So no borders anymore.

HDinger commented:
Hi @wielinde 

thanks for the changes. 👍 I have only one minor thing left: When resizing a widget, the other ones are very hard to see. While dragging, they have a border. So maybe, we could use the same style while resizing? In the best case, the classes `-dragged` and `-passive` share the same rules.
![Bildschirmfoto 2019-10-15 um 13 52 09 (2)](https://user-images.githubusercontent.com/7457313/66829430-acaf3b80-ef53-11e9-9b14-3ceac15572a7.png)

wielinde committed:
Unify styles for passive widgets while dragging or resizing [ci skip]

wielinde commented:
@HDinger I rebased to latest `dev` and "force pushed" my changes regarding the styles of the passive widgets during resizing or dragging.

HDinger reviewed (approved):
👍
